nomos-c/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+// Safety: Well, this is gonna be a shit show of unsafe calls...
+#![allow(clippy::allow_attributes_without_reason)]
+#![allow(clippy::undocumented_unsafe_blocks)]
+
+mod api;
+
+use std::ffi::c_char;
+
+pub use api::{stop_node, NomosNode};
+use nomos_node::{
+    get_services_to_start, AdapterSettings, Config, Nomos, NomosServiceSettings, SignedMantleTx,
+    SignedTxProcessorSettings, Transaction, TxMempoolSettings,
+};
+use overwatch::overwatch::OverwatchRunner;
+use tokio::runtime::Runtime;
+
+#[repr(u8)]
+pub enum NomosNodeErrorCode {
+    None = 0x0,
+    CouldNotInitialize = 0x1,
+    StopError = 0x2,
+    NullPtr = 0x3,
+}
+
+#[repr(C)]
+pub struct InitializedNomosNodeResult {
+    nomos_node: *mut NomosNode,
+    error_code: NomosNodeErrorCode,
+}
+
+#[unsafe(no_mangle)]
+pub extern "C" fn start_nomos_node(config_path: *const c_char) -> InitializedNomosNodeResult {
+    match initialize_nomos_node(config_path) {
+        Ok(nomos_node) => {
+            let node_ptr = Box::into_raw(Box::new(nomos_node));
+            InitializedNomosNodeResult {
+                nomos_node: node_ptr,
+                error_code: NomosNodeErrorCode::None,
+            }
+        }
+        Err(error_code) => InitializedNomosNodeResult {
+            nomos_node: core::ptr::null_mut(),
+            error_code,
+        },
+    }
+}
+
+fn initialize_nomos_node(config_path: *const c_char) -> Result<NomosNode, NomosNodeErrorCode> {
+    let must_blend_service_group_start = true;
```

Comment:
This would be changed with a CLI flag, ideally, so perhaps they should be inputs to the `initialize_nomos_node` function?

Reply:
This will have to change shortly when we do it dynamically in the node (?). So probably not worth it to expose it.

Reply:
Ok makes sense. Then perhaps we should move them waaay below just before they are used. At least that is what I would prefer, but if you prefer to keep them at the top, let's leave it as it is.

Reply:
I prefer to have them on top just so its obvious that they will be removed at some point. I'll add a TODO to it as well.